### PR TITLE
JDK-8308028 Replace more uses of System.getProperty("os.name") with PlatformUtil calls

### PR DESCRIPTION
**Pull request by EstelonAgarwaen**
replaced implementation of OS checks in HostUtils with calls of PlatformUtil

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8308028](https://bugs.openjdk.org/browse/JDK-8308028): Replace more uses of System.getProperty("os.name") with PlatformUtil calls

### Reviewers
 * [Nir Lisker](https://openjdk.org/census#nlisker) (@nlisker - **Reviewer**)
 * [Marius Hanl](https://openjdk.org/census#mhanl) (@Maran23 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1134/head:pull/1134` \
`$ git checkout pull/1134`

Update a local copy of the PR: \
`$ git checkout pull/1134` \
`$ git pull https://git.openjdk.org/jfx.git pull/1134/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1134`

View PR using the GUI difftool: \
`$ git pr show -t 1134`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1134.diff">https://git.openjdk.org/jfx/pull/1134.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1134#issuecomment-1546699432)

### CODE REVIEW

File: modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/HostUtils.java [hunk 1]
```diff
@@ -57,19 +59,19 @@ public static boolean is64Bit() {
...
-        return osName.startsWith("linux");
+        return PlatformUtil.isLinux();
     }
 
     public static boolean isIOS() {
-        return osName.startsWith("ios");
+        return PlatformUtil.isIOS();
     }
 
     /**
```
Comment (Maran23) [inline modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/HostUtils.java:77]: `isEmbedded` can also be replaced by `PlatformUtil`.

File: modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/HostUtils.java [hunk 2]
```diff
@@ -57,19 +59,19 @@ public static boolean is64Bit() {
     }
 
     public static boolean isWindows() {
-        return osName.startsWith("windows");
```
Comment (Maran23) [inline modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/HostUtils.java:60]: `osName` is unused now

Comment (nlisker) [reply modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/HostUtils.java:60]: `osArch` is also unused. It's just a local variable.

Comment (EstelonAgarwaen) [reply modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/HostUtils.java:60]: @nlisker osArch seems to be used for is64bit still

Comment (Maran23) [reply modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/HostUtils.java:60]: Yes, but what @nlisker means is that `osArch` does not need to be a member variable but only a local one inside the static initializer above.

**Comment by EstelonAgarwaen** [comment]
I have looked at `HostUtils` further. The only method it implements itself is the `is64bit`, which checks whether the architechture is 64 bit. i could not find any uses os this method anywhere. 

i think it may be a valid option to replace calls of `HostUtils` by calls of `PlatformUtil` and move `is64bit` in the `PlatformUtil` as well. It may also be an option to remove it completely.

File: modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/HostUtils.java [hunk 3]
```diff
@@ -25,57 +25,53 @@
...
-    private static boolean is64bit = false;
+    private static boolean is64bit;
 
     static {
         @SuppressWarnings("removal")
-        boolean tmp = AccessController.doPrivileged((PrivilegedAction<Boolean>) () -> {
-            osName = System.getProperty("os.name").toLowerCase();
-            osArch = System.getProperty("os.arch").toLowerCase();
+        boolean temp = AccessController.doPrivileged((PrivilegedAction<Boolean>) () -> {
+            String osArch = System.getProperty("os.arch").toLowerCase();
```
Comment (Glavo) [inline modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/HostUtils.java:42]:
```suggestion
            String osArch = System.getProperty("os.arch").toLowerCase(Locale.ROOT);
```

**Comment by Glavo** [comment]
> I have looked at `HostUtils` further. The only method it implements itself is the `is64bit`, which checks whether the architechture is 64 bit. i could not find any uses os this method anywhere.
> 
> i think it may be a valid option to replace calls of `HostUtils` by calls of `PlatformUtil` and move `is64bit` in the `PlatformUtil` as well. It may also be an option to remove it completely.

Sounds great, I was just about to say the same.

**Comment by Maran23** [comment]
> i think it may be a valid option to replace calls of `HostUtils` by calls of `PlatformUtil` and move `is64bit` in the `PlatformUtil` as well. It may also be an option to remove it completely.

Since `is64bit` is unused, `HostUtils` is basically a delegate for `PlatformUtil`. So we might as well delete it completely and replace all occurrences with `PlatformUtil`. And as written by Kevin in the ticket, this is a valid choice so we might as well do it.

**Comment by nlisker** [comment]
`isEmbedded()` is also unused as far as I can tell.

The question is why this and `is64bit` were created. Before removing unused code I would try to find out if it is *supposed* to be used.

> i think it may be a valid option to replace calls of HostUtils by calls of PlatformUtil and move is64bit in the PlatformUtil as well. It may also be an option to remove it completely.

This is what I would have done. No need to have duplicate functionalities.

**Comment by EstelonAgarwaen** [comment]
Since `HostUtils` is internal API and not used outside of JFX itself, and `is64bit` is not used anywhere i would say its safe to remove. 

From its implementation I saw that it checked whether the operating systems architechture was either intel/amd 64 bit or intel itanium(ia64). In any case of an ARM-Architecture, even on ARM64, the method would have returned false.

**Review by nlisker — APPROVED** [review]

**Review by Maran23 — APPROVED** [review]
Looks good.

**Pull request closed without merging**